services/intersection-api/api/src/main/java/us/dot/its/jpo/ode/api/accessors/map/ProcessedMapRepositoryImpl.java

```diff
@@ -0,0 +1,289 @@
+import us.dot.its.jpo.geojsonconverter.pojos.geojson.LineString;
+
+@Component
+public class ProcessedMapRepositoryImpl implements ProcessedMapRepository {
+
+    @Autowired
+    private MongoTemplate mongoTemplate;
+
+    @Autowired
+    ConflictMonitorApiProperties props;
+
+    TypeReference<ProcessedMap<LineString>> processedMapTypeReference = new TypeReference<>(){};
+
+    private String collectionName = "ProcessedMap";
+    private ObjectMapper mapper = DateJsonMapper.getInstance();
+    private Logger logger = LoggerFactory.getLogger(ProcessedMapRepositoryImpl.class);
+
+    public Query getQuery(Integer intersectionID, Long startTime, Long endTime, boolean latest, boolean compact) {
+        Query query = new Query();
+
+        if (intersectionID != null) {
+            query.addCriteria(Criteria.where("properties.intersectionId").is(intersectionID));
+        }
+
+        String startTimeString = Instant.ofEpochMilli(0).toString();
+        String endTimeString = Instant.now().toString();
+
+        if (startTime != null) {
+            startTimeString = Instant.ofEpochMilli(startTime).toString();
+        }
+        if (endTime != null) {
+            endTimeString = Instant.ofEpochMilli(endTime).toString();
+        }
+
+        if (latest) {
+            query.with(Sort.by(Sort.Direction.DESC, "properties.timeStamp"));
+            query.limit(1);
+        }else{
+            query.limit(props.getMaximumResponseSize());
+        }
+
+        if (compact){
+            query.fields().exclude("recordGeneratedAt", "properties.validationMessages");
+        }else{
+            query.fields().exclude("recordGeneratedAt");
+        }
+
+        query.addCriteria(Criteria.where("properties.timeStamp").gte(startTimeString).lte(endTimeString));
+        return query;
+    }
+
+    public long getQueryResultCount(Query query) {
+        return mongoTemplate.count(query, ProcessedMap.class, collectionName);
+    }
+
+    public long getQueryFullCount(Query query){
+        int limit = query.getLimit();
+        query.limit(-1);
+        long count = mongoTemplate.count(query, ProcessedMap.class, collectionName);
+        query.limit(limit);
+        return count;
+    }
+
+    public List<ProcessedMap<LineString>> findProcessedMaps(Query query) {
+        List<Map> documents = mongoTemplate.find(query, Map.class, collectionName);
+        List<ProcessedMap<LineString>> convertedList = new ArrayList<>();
+        for (Map document : documents) {
+            document.remove("_id");
+            ProcessedMap<LineString> bsm = mapper.convertValue(document, processedMapTypeReference);
+            convertedList.add(bsm);
+        }
+        return convertedList;
+    }
+
+    public List<IntersectionReferenceData> getIntersectionIDs() {
+        MongoCollection<Document> collection = mongoTemplate.getCollection(collectionName);
+        DistinctIterable<Integer> docs = collection.distinct("properties.intersectionId", Integer.class);
```

Comment:
question: would it be more efficient to do this all in Mongo instead of pulling parts of the data back and re-querying with it? ex:

- use a $group to group by intersection id
- sort by timeStamp
- get first object in each group
- project properties we want

Reply:
Created a task item to resolve this in the future: https://dev.azure.com/SOC-OIT/CDOT/_backlogs/backlog/CDOT%20Connected%20Vehicles/Backlog%20items/?workitem=289476